src/jit/codegencommon.cpp

```diff
@@ -7369,10 +7398,18 @@ void CodeGen::genProfilingEnterCallback(regNumber initReg, bool* pInitRegZeroed)
     // Can't have a call until we have enough padding for rejit
     genPrologPadForReJit();
 
+#if !defined(UNIX_AMD64_ABI)
     // This will emit either
     // "call ip-relative 32-bit offset" or
     // "mov rax, helper addr; call rax"
     genEmitHelperCall(CORINFO_HELP_PROF_FCN_ENTER, 0, EA_UNKNOWN);
+#else
+    // We can use any callee trash register (other than RAX, RDI, RSI) for call target.
+    // We use R11 here. This will emit either
+    // "call ip-relative 32-bit offset" or
+    // "mov r11, helper addr; call r11"
+    genEmitHelperCall(CORINFO_HELP_PROF_FCN_ENTER, 0, EA_UNKNOWN, REG_R11);
```

Comment:
@sivarv so you are saying that whenever JIT emits the Enter callback, it would also have to ensure that all argument registers are homed in the callee frame at a well known location relative to the frame start on the stack that the profiler can access, right?
While it would work that way too, it would also require a codegen difference due to the different location of the homed arguments. 
The way @sergign60 has implemented it is slightly different - instead of relying on the homing, we rely on the argument registers being passed unchanged to the Enter callback. And since the RDI and RSI are needed for the Enter callback arguments, he is saving them to R12 and R13 before setting  RDI and RSI.